src/Apps/Authentication/Utils/useAuthForm.ts

```diff
@@ -12,45 +17,49 @@ interface UseAuthFormProps {
   type: ModalType
 }
 
-export function useAuthForm({
+export const useAuthForm = ({
   canonical,
   description,
   pageTitle,
   type,
-}: UseAuthFormProps) {
+}: UseAuthFormProps) => {
   const { match } = useRouter()
 
   const {
-    action,
     afterSignUpAction,
     contextModule,
     copy: copyQueryParam,
     destination,
     intent,
-    kind,
     oauthLogin,
-    objectId,
-  } = match.location.query as ModalOptions
-
-  const title = computeTitle({
-    copy: copyQueryParam,
-    intent,
-    pageTitle,
-  })
+    // FIXME: Convection should link using the `afterSignUpAction` param, not `submissionId`
+    submissionId,
+  } = match.location.query as ModalOptions & { submissionId?: string }
 
+  const title = computeTitle({ copy: copyQueryParam, intent, pageTitle })
   const redirectTo = getENV("AUTHENTICATION_REDIRECT_TO")
   const signupReferer = getENV("AUTHENTICATION_REFERER")
 
-  if (action) {
-    const isClient = typeof window !== "undefined"
+  // FIXME: Convection should link using the `afterSignUpAction` param,
+  // not `submissionId` so that we don't have to do this.
+  //
+  // If there's a `submissionId` constructs an `afterSignUpAction` from it. Otherwise
+  // falls back to the `afterSignUpAction` param.
+  const _afterSignUpAction: AfterAuthAction | undefined = !!submissionId
+    ? {
+        action: "associateSubmission",
+        kind: "submission",
+        objectId: submissionId,
+      }
+    : afterSignUpAction
 
-    if (isClient) {
-      Cookies.set(
-        "afterSignUpAction",
-        JSON.stringify({ action, kind, objectId })
-      )
-    }
-  }
+  // FIXME: We have to make the cookie now because apparently passing it into
+  // options like this doesn't even work!
```

Comment:
I will revisit all of this after the rebuild is done and I need to integrate the new form into the static routes, where this is an issue.